### PR TITLE
Add patch to fix linking with python in macOS

### PR DESCRIPTION
**Pull request by tadeu**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] ~~Ensured the license file is being packaged.~~

### CODE REVIEW

**Comment by tadeu** [comment]
@conda-forge-admin, please rerender

**Comment by tadeu** [comment]
@conda-forge-admin, please rerender

File: recipe/0001-Fix-linking-with-python-in-macOS.patch [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+@@ -12,7 +12,11 @@ if (POLICY CMP0042)
+   cmake_policy (SET CMP0042 NEW)
+ endif()
+ 
+-option ( LINK_PYTHON_LIBRARY "Link against python libraries" ON )
++if ( APPLE )
++  option ( LINK_PYTHON_LIBRARY "Link against python libraries - NOT RECOMMENDED" OFF )
++else ()
++  option ( LINK_PYTHON_LIBRARY "Link against python libraries" ON )
++endif ()
```
Comment (lrineau) [inline recipe/0001-Fix-linking-with-python-in-macOS.patch]: @tadeu I have asked a question in the upstream project: https://github.com/CGAL/cgal-swig-bindings/pull/114#pullrequestreview-217370631.

**Comment by tadeu** [comment]
Fixes #46

**Pull request merged**